### PR TITLE
feat: W3C verifcation with connections.

### PR DESCRIPTION
**Pull request by pranalidhanavade**
### What

- Refactor existing flow of verification with connections.
- Add payload structure which will be work for both INDY and W3C schemaTypes.

### CODE REVIEW

File: src/commonComponents/interface.ts [hunk 1]
```diff
@@ -25,4 +25,5 @@ export interface ISchemaCardProps {
 	limitedAttributes?: boolean
 	w3cSchema:boolean
 	noLedger:boolean
+	isVerification:boolean
```
Comment (bhavanakarwade) [inline src/commonComponents/interface.ts:28]: Is `isVerification` property is mandatory field?

File: src/components/Verification/interface.ts [hunk 2]
```diff
@@ -60,6 +60,21 @@ export interface SchemaDetail {
 	schemaId: string;
 	credDefId: string;
 }
+export interface IW3cSchemaDetails {
+	schemaName: string;
+	version: string;
```
Comment (bhavanakarwade) [inline src/components/Verification/interface.ts:65]: Is `version` is required for W3C type schema

File: src/components/Verification/Verification.tsx [hunk 3]
```diff
@@ -31,13 +34,64 @@ const VerificationCred = () => {
...
+
+				}
+				if (did?.includes(DidMethod.INDY)) {
+					setW3cSchema(false);
+					setRequestType(RequestType.indy)
+					await getSchemaAndUsers(false);
+				}
+			}
+		} catch(error){
+			console.log('Error in getSchemaAndUsers:', error);
```
Comment (bhavanakarwade) [inline src/components/Verification/Verification.tsx:88]: Remove this console.log

**Review by bhavanakarwade — CHANGES_REQUESTED** [review]

File: src/components/Verification/interface.ts (hunk 2 above)
Comment (pranalidhanavade) [reply src/components/Verification/interface.ts:65]: Yes, It is required to display schema version on schema card.

File: src/components/Verification/Verification.tsx (hunk 3 above)
Comment (pranalidhanavade) [reply src/components/Verification/Verification.tsx:88]: Yes.

**Review by bhavanakarwade — APPROVED** [review]
LGTM

File: src/common/enums.ts [hunk 4]
```diff
@@ -141,3 +141,13 @@ export enum CredentialType {
     INDY = 'indy',
     JSONLD = 'jsonld'
 }
+
+export enum RequestType {
+    indy = 'indy',
```
[the remark this answers is not in the document]
Comment (pranalidhanavade) [reply src/common/enums.ts:146]: Yes.

File: src/commonComponents/SchemaCard.tsx [hunk 5]
```diff
@@ -124,17 +124,40 @@ const SchemaCard =  (props: ISchemaCardProps) => {
...
+      {props.isVerification && props.w3cSchema &&(
+          <div className="p-2">
+            <Button
+          onClick={() => {
+            handleIssueClick();
+            window.location.href = pathRoutes.organizations.verification.requestProof;
+          }}
+              type="submit"
+              color='bg-primary-800'
+              title='Initiate Credential Issuance'
```
Comment (tipusinghaw) [inline src/commonComponents/SchemaCard.tsx:145]: Title of this verification button is `Initiate Credential Issuance'` , is it intentional ?

File: src/commonComponents/SchemaCard.tsx [hunk 6]
```diff
@@ -124,17 +124,40 @@ const SchemaCard =  (props: ISchemaCardProps) => {
...
-        )}
-        </div>
-      </div>
+          )}
+
+      {props.isVerification && props.w3cSchema &&(
+          <div className="p-2">
+            <Button
+          onClick={() => {
+            handleIssueClick();
```
Comment (tipusinghaw) [inline src/commonComponents/SchemaCard.tsx:140]: Why we are using `handleIssueClick` function in verification flow ?

File: src/components/Issuance/IssuedCrdentials.tsx [hunk 7]
```diff
@@ -45,6 +45,8 @@ const CredentialList = () => {
 		nextPage: '',
 		lastPage: '',
 	});
+	const [w3cSchema, setW3cSchema]= useState<boolean>(false);
```
Comment (tipusinghaw) [inline src/components/Issuance/IssuedCrdentials.tsx:48]:
```suggestion
	const [w3cSchema, setW3CSchema]= useState<boolean>(false);
```

File: src/components/Resources/Schema/Create.tsx [hunk 8]
```diff
@@ -94,7 +94,11 @@ const CreateSchema = () => {
 
 	const areAllInputsFilled = (formData: IFormData) => {
```
Comment (tipusinghaw) [inline src/components/Resources/Schema/Create.tsx:95]: Think we should change the function name

File: src/components/Verification/Verification.tsx [hunk 9]
```diff
@@ -160,18 +214,56 @@ const VerificationCred = () => {
...
+				if(w3cSchema) { 
+
+					const attributePaths = attributes?.map(
+						(attr) => `$.credentialSubject['${attr.attributeName}']`
+					  );
+					verifyCredentialPayload = {
+						connectionId: JSON.parse(userData)[0]?.connectionId,
+						comment: 'proof request',
+						presentationDefinition: {
+						  id: '32f54163-7166-48f1-93d8-ff217bdb0653',
```
Comment (tipusinghaw) [inline src/components/Verification/Verification.tsx:241]: Why we are using static ID value

File: src/components/Verification/Verification.tsx [hunk 10]
```diff
@@ -246,7 +371,7 @@ const VerificationCred = () => {
 							),
 						},
 						{ data: attribute?.displayName },
-						{
+						!w3cSchema && {
```
Comment (tipusinghaw) [inline src/components/Verification/Verification.tsx:374]: Why we are using same condition at line 396

File: src/components/Verification/Verification.tsx (hunk 9 above)
Comment (pranalidhanavade) [reply src/components/Verification/Verification.tsx]: Resolved.

File: src/commonComponents/SchemaCard.tsx (hunk 6 above)
Comment (pranalidhanavade) [reply src/commonComponents/SchemaCard.tsx]: Resolved.

File: src/commonComponents/SchemaCard.tsx (hunk 5 above)
Comment (pranalidhanavade) [reply src/commonComponents/SchemaCard.tsx]: Resolved.

File: src/components/Resources/Schema/Create.tsx (hunk 8 above)
Comment (pranalidhanavade) [reply src/components/Resources/Schema/Create.tsx:95]: Resolved.

File: src/components/Verification/Verification.tsx (hunk 10 above)
Comment (pranalidhanavade) [reply src/components/Verification/Verification.tsx:375]: Both code blocks are different and requires separate conditions.

File: src/components/Issuance/IssuedCrdentials.tsx (hunk 7 above)
Comment (pranalidhanavade) [reply src/components/Issuance/IssuedCrdentials.tsx:48]: Resolved.

**Review by tipusinghaw — APPROVED** [review]

**Pull request merged**